app/main.py

```diff
@@ -1 +1,59 @@
-# write your code here
+class SoftwareEngineer:
+
+    def __init__(self, name: str):
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str):
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills.extend([
+            "JavaScript",
+            "HTML",
+            "CSS",
+        ])
+
+    def create_awesome_web_page(self):
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills.extend([
+            "Python",
+            "SQL",
+            "Django",
+        ])
+
+    def create_powerful_api(self):
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills.extend([
+            "Java",
+            "Android studio",
+        ])
+
+    def create_smooth_mobile_app(self):
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(FrontendDeveloper, BackendDeveloper,):
+    def __init__(self, name: str):
```

Comment:
if the constructor is absent in class the parent's one is called by default 

so this constructor is extra

Reply:
understand, thank you!